lib/scenic/script.ex

```diff
@@ -534,6 +539,108 @@ defmodule Scenic.Script do
     [{:draw_rrect, {width, height, radius, flag}} | ops]
   end
 
+  @doc """
+  Draw a rounded rectangle defined by height, width, radius1 and radius2. Can be filled or stroked.
+
+  Radius1 and radius2 values will be set as follow:
+
+  - Upper left corner: radius1
+  - Upper right corner: radius2
+  - Lower right corner: radius1
+  - Lower left corner: radius2
+
+  Creates a new path and draws it.
+  """
+  @spec draw_rounded_rectangle(
+          ops :: t(),
+          width :: number,
+          height :: number,
+          r1 :: number,
+          r2 :: number,
+          fill_stroke_flags :: fill_stroke()
+        ) :: ops :: t()
+  def draw_rounded_rectangle(ops, width, height, r1, r2, flag) do
+    upperLeftRadius = smallest([r1, width / 2, height / 2])
+    upperRightRadius = smallest([r2, width / 2, height / 2])
+    lowerRightRadius = smallest([r1, width / 2, height / 2])
+    lowerLeftRadius = smallest([r2, width / 2, height / 2])
+
+    [
+      {:draw_rrectv,
+       {width, height, upperLeftRadius, upperRightRadius, lowerRightRadius, lowerLeftRadius, flag}}
+      | ops
+    ]
+  end
+
+  @doc """
+  Draw a rounded rectangle defined by height, width, radius1, radius2 and radius3. Can be filled or stroked.
+
+  Radius1 and radius2 values will be set as follow:
+
+  - Upper left corner: radius1
+  - Upper right corner: radius2
+  - Lower right corner: radius3
+  - Lower left corner: radius2
+
+  Creates a new path and draws it.
+  """
+  @spec draw_rounded_rectangle(
+          ops :: t(),
+          width :: number,
+          height :: number,
+          r1 :: number,
+          r2 :: number,
+          r3 :: number,
+          fill_stroke_flags :: fill_stroke()
+        ) :: ops :: t()
+  def draw_rounded_rectangle(ops, width, height, r1, r2, r3, flag) do
+    upperLeftRadius = smallest([r1, width / 2, height / 2])
+    upperRightRadius = smallest([r2, width / 2, height / 2])
+    lowerRightRadius = smallest([r3, width / 2, height / 2])
+    lowerLeftRadius = smallest([r2, width / 2, height / 2])
+
+    [
+      {:draw_rrectv,
+       {width, height, upperLeftRadius, upperRightRadius, lowerRightRadius, lowerLeftRadius, flag}}
+      | ops
+    ]
+  end
```

Comment:
I like this idea, but I'd suggest that we limit this API to the `/8` variant to keep things cleaner.

Reply:
I agree. This is in the scripting API, so ok to just have the single variant that should cover all the cases. Lets just keep the /8 version.

Reply:
Just to remind, keeping only the /8 version will introduce breaking changes to `draw_rounded_rectangle/5`.
Or we will use default arguments?

Reply:
This should be a new function. draw_variable_rounded_rectangle with a new opcode. I don't want the wire format to change between scenic and the drivers. So the old parser should still work with existing data. This is just additive.

Reply:
Perfect @boydm!
On `scenic_driver_local` this is already a new opcode, so the only part to change is here.
Will fix this asap.

Reply:
Great!